### PR TITLE
Get Watermark, Apply Watermark, Remove Watermark on folder, file

### PR DESCRIPTION
**Pull request by marek-vysoky**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @marek-vysoky has signed the CLA. Thanks for the pull request!

File: Box.V2/Models/Request/BoxApplyWatermarkRequest.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Box.V2.Models
+{
+    public class BoxApplyWatermarkRequest
+    {
+        private BoxWatermarkRequest watermark = new BoxWatermarkRequest();
```
Comment (coolboy) [inline Box.V2/Models/Request/BoxApplyWatermarkRequest.cs]: @cburnette  is this the right way of embedding?

File: Box.V2/Managers/BoxFilesManager.cs [hunk 2]
```diff
@@ -667,6 +667,81 @@ private int GetTimeDelay(HttpResponseHeaders headers)
...
+            {
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Used to apply or update the watermark for a corresponding Box file.
+        /// </summary>
+        /// <param name="id">Id of the file.</param>
+        /// <param name="applyWatermarkRequest">The apply watermark request. Can be null, for using default values - imprint="default" </param>
```
Comment (coolboy) [inline Box.V2/Managers/BoxFilesManager.cs]: update the comment to BoxApplyWatermarkRequest object.

File: Box.V2/Managers/BoxFoldersManager.cs [hunk 3]
```diff
@@ -358,5 +358,80 @@ public BoxFoldersManager(IBoxConfig config, IBoxService service, IBoxConverter c
...
+            {
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Used to apply or update the watermark for a corresponding Box folder.
+        /// </summary>
+        /// <param name="id">Id of the folder.</param>
+        /// <param name="applyWatermarkRequest">The apply watermark request. Can be null, for using default values - imprint="default" </param>
```
Comment (coolboy) [inline Box.V2/Managers/BoxFoldersManager.cs]: update the comment to BoxApplyWatermarkRequest object.

Comment (marek-vysoky) [inline Box.V2/Managers/BoxFoldersManager.cs]: @coolboy what do you think?

File: Box.V2/Models/Request/BoxApplyWatermarkRequest.cs (hunk 1 above)
Comment (cburnette) [inline Box.V2/Models/Request/BoxApplyWatermarkRequest.cs]: hmm, reading through all this now.  I'm going to need to pull this PR down and take a look at it in detail

**Comment by cburnette** [comment]
@coolboy I'm going through this now in a new branch so i can make a few changes

**Pull request merged**